lang/funcs/core/net/mac_fmt_func_test.go

```diff
@@ -0,0 +1,52 @@
+// Mgmt
+// Copyright (C) 2013-2020+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package corenet
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/purpleidea/mgmt/lang/types"
+)
+
+func TestMacFmt(t *testing.T) {
+	var tests = []struct {
+		name    string
+		in      string
+		out     string
+		wantErr bool
+	}{
+		{"Valid mac with hyphens", "01-23-45-67-89-AB", "01:23:45:67:89:AB", false},
+		{"Valid mac with colons", "01:23:45:67:89:AB", "01:23:45:67:89:AB", false},
+		{"Invalid mac", "", "", true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
t.Run can contain a goroutine, so beware the bug you've introduced:

https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables

Reply:
Added `tt := tt` to mitigate the bug as recommended in the link.